### PR TITLE
Add ARB

### PR DESCRIPTION
**Pull request by epruesse**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Adds phylogenetic software suite [ARB](http://www.arb-home.de). The package name is `arb-bio` (not `arb` as e.g. in Debian) because `arb` was already taken. 

This package is currently missing 
* [x] lynx (PR conda-forge/staged-recipes#3123)
* [x] time (PR conda-forge/staged-recipes#3139)

### CODE REVIEW

**Comment by epruesse** [comment]
Muahaha! This was one difficult package! But it's working now! :-D
@bioconda/core Anyone have time for a review?

**Comment by daler** [comment]
Wow. @epruesse this is amazing!

The only comment I have is that you have preprocessing selectors in meta.yaml for Windows which we don't build for. But I almost hesitate to say anything, since including them implies you also went through the headache of testing on Windows.

Impressive work!

**Comment by epruesse** [comment]
@daler Do you mean the `# [not win]`? I think that was copy&paste. I did some windows packages and I like the `conda inspect` to tell me what's being linked. The mulled test will fail if I forget something, but I don't see if I could lose a requirement.

Getting ARB to work on windows would be a major project, if it can be done at all.

File: recipes/arb-bio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+   # prefix SOURCE_TOOLS/... with full PATH
+   - make_path_fix.patch
+   # change all /usr/bin/xyz paths to /usr/bin/env xyz
+   - abs_path_fix.patch
+   # use sed, not gsed
+   - gsed.patch
+
+build:
+  number: 0
+  skip: True  # [win]
```
Comment (daler) [inline recipes/arb-bio/meta.yaml]: Can delete this line

File: recipes/arb-bio/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    - sed >=4.4
+    - xfig
+    - fig2dev
+    - perl !=5.22.0.1-0,!=5.22.2.1-0
+
+test:
+  requires:
+    - conda
+  commands:
+    - if command -v conda; then conda inspect linkages -p $PREFIX $PKG_NAME; fi  # [not win]
```
Comment (daler) [inline recipes/arb-bio/meta.yaml]: `s/not win/linux/`

**Comment by daler** [comment]
You're right, mulled-build will let you know of any issues. This is more of a style thing or to avoid future confusion of recipe updaters/maintainers.

File: recipes/arb-bio/meta.yaml (hunk 2 above)
Comment (epruesse) [inline recipes/arb-bio/meta.yaml]: Just removed it, it'll work on both Linux and OSX.

File: recipes/arb-bio/meta.yaml (hunk 1 above)
Comment (epruesse) [inline recipes/arb-bio/meta.yaml]: Done

**Comment by epruesse** [comment]
This build will fail though, PhyML is merged but not built yet, I was too quick on that one.

**Comment by daler** [comment]
OK. This looks good to me otherwise; feel free to merge when it's ready and thanks again.

**Comment by bgruening** [comment]
Cool! Glad it worked in the end! Thanks @epruesse!

**Comment by epruesse** [comment]
Thanks guys. It took a while to get here. See epruesse/SINA#8 for a list of things that were needed before this recipe could go in. ;*)

I may have to fiddle with it once I can run it through its paces on an X11. Testing whether all the integrations (tree reconstruction and alignment tools) are working right requires a display, and prepping that on Travis was too much for me for the time being. Anyway, it's possible that ARB needs different versions of those tools. It usually comes with them, but in a distro they should be separate packages (with pinning if needed), so I disabled them and built separately.

**Pull request merged**